sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/storage/DatabaseManager.java

```diff
@@ -310,18 +323,18 @@ public final long getRowCount() {
      * Gets a cursor for all rows in the table, all rows where key matches value if specified.
      *
      * @param queryBuilder  The query builder that contains SQL query.
+     * @param columns       Columns to select, null for all.
      * @param selectionArgs The array of values for selection.
-     * @param idOnly        Return only row identifier if true, return all fields otherwise.
+     * @param sortOrder     Sorting order (ORDER BY clause without ORDER BY itself).
      * @return A cursor for all rows that matches the given criteria.
      * @throws RuntimeException If an error occurs.
      */
-    public Cursor getCursor(@Nullable SQLiteQueryBuilder queryBuilder, @Nullable String[] selectionArgs, boolean idOnly) throws RuntimeException {
+    public Cursor getCursor(@Nullable SQLiteQueryBuilder queryBuilder, String[] columns, @Nullable String[] selectionArgs, @Nullable String sortOrder) throws RuntimeException {
```

Comment:
During implementation at this commit I needed the new signature because I was selecting only columns oid and priority to avoid loading all data: https://github.com/Microsoft/AppCenter-SDK-Android/pull/858/commits/19f6c5a8fc58d269061b85190c50d6fbd2d01151#diff-9b79a28544bf85990330f7945382e4ebR217

Later I simplified code and now need only id like before. But I kept the refactoring change as it also allows to optimize the count query, which is now included in this change, this could have been part of the storagehelper refactoring but didn't catch that during review.